sql/external_stored_procedure_registry.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import (
+	"math"
+	"reflect"
+	"strings"
+)
+
+// ExternalStoredProcedureRegistry manages a collection of ExternalStoredProcedures and encapsulates
+// the logic for looking up external stored procedures based on name and number of arguments.
+type ExternalStoredProcedureRegistry struct {
+	procedures map[string]map[int]ExternalStoredProcedureDetails
+}
+
+// NewExternalStoredProcedureRegistry creates a new, empty instance of ExternalStoredProcedureRegistry.
+func NewExternalStoredProcedureRegistry() ExternalStoredProcedureRegistry {
```

Comment:
I would move this type to the memory package

Reply:
I see this is used by Dolt's implementation, fine leaving it where it is

Reply:
Yup, you figured it out. I didn't want to copy/paste that code in both providers. I almost threw it in a new `sql.util` package or something to avoid cluttering `sql` but figured this was the right place to start it off.